### PR TITLE
Conversion to SCSS

### PR DESCRIPTION
**Pull request by dimaursu**
I used a convertor to exact the colors into a separate file. The names would need a bit of work, I would like them to convey the way they will be used: for example not $red, but $color_status_offline, something like that. We would also need some sort of convention, to make the change consistent. 

I also added some Grunt plugins, including a CSS linter, Autoprefixer to forget about adding vendor prefixes, and a new "css" task, to avoid buliding the whole project.

The final destination of the css files is same as previous, but at the build stage they are separated into `css` and `scss` folders. One would edit just the `scss` files. I'm not sure it's a good idea to version `css` at all.

### CODE REVIEW

**Comment by Flaburgan** [comment]
I think this is the kind of things we want to see merged upstream. To maintain this patch in our repo would imply to manually merge and convert each time upstream changes some CSS. Could you open your pull request on https://github.com/sualko/jsxc?

**Comment by dimaursu** [comment]
I will separate the commits in a separate rebased branch. Right now I pull too much stuff in the pull-request alongside the SCSS work.

**Comment by dimaursu** [comment]
Also, if I think about it, we will probably customize the interface much more, to integrate it into Diaspora -> that means our CSS will grow apart from the @sualko's; Leaving the visual identity to that repository is no good, imho.

File: jsxc.lib.js [hunk 1]
```diff
@@ -4530,10 +4485,6 @@ var jsxc;
             jsxc.xmpp.addBuddy(n.username, n.alias);
          }*/
 
-         if (key === 'roster') {
-            jsxc.gui.roster.toggle();
```
Comment (Zauberstuhl) [inline jsxc.lib.js]: that is not the only time `toggle` is called

File: jsxc.lib.js [hunk 2]
```diff
@@ -3034,9 +2989,12 @@ var jsxc;
                 </div>\
             </div>\
         </li>',
-      roster: '<div id="jsxc_roster">\
-           <ul id="jsxc_buddylist"></ul>\
-           <div class="jsxc_bottom jsxc_presence" data-bid="own">\
+      roster: '<input type="checkbox" id="jsxc_toggleRoster_text">\
+          <div id="jsxc_roster">\
+            <label id="jsxc_toggleRoster" for="jsxc_toggleRoster_text"></label>\
+            <span class="entypo"></span>\
```
Comment (Zauberstuhl) [inline jsxc.lib.js]: on my test setup it is displayed in the roster list instead of the `jsxc_toggleRoster`

File: jsxc.lib.js (hunk 1 above)
Comment (Zauberstuhl) [inline jsxc.lib.js]: fixed witgh #76

**Comment by Zauberstuhl** [comment]
@dimaursu you have to rebase against develop

**Comment by jhass** [comment]
I agree with Fla, we should not merge this, upstream should

**Comment by dimaursu** [comment]
@sualko said he is interested. I'll wait untill I have more feedback from him.
https://github.com/sualko/jsxc/issues/118

**Pull request merged**